src/services/errorService.js

```diff
@@ -14,96 +14,57 @@ See the License for the specific language governing permissions and
 limitations under the License. */
 
 import * as Sentry from '@sentry/react';
-import networkService from 'services/networkService';
-import { getToken } from 'actions/tokenAction';
-import { logAmount } from 'util/amountConvert';
-import JSON5 from 'json5';
 import config from 'util/config';
 
 if (config.sentry) {
   Sentry.init({ dsn: config.sentry });
 }
 
-// skip error logging
+const errorCache = [];
 const noLogErrors = [
   'user denied',
   'user rejected',
-  'server is not ready to handle the request',
-  'insufficient funds',
-  'network error',
-  'transaction has been reverted'
+  'user canceled'
 ];
 
-// used to prevent error spam from a single session
-const errorCache = [];
-
-class ErrorService {
-  log (error) {
-    // skip logging if in defined list
-    for (const i of noLogErrors) {
-      if (error.message.toLowerCase().includes(i)) {
-        return;
-      }
-    }
-
-    if (config.sentry) {
-      if (errorCache.includes(error.message)) {
-        // if message already seen in this session, ignore it
-        return;
-      }
-      errorCache.push(error.message);
-      Sentry.captureException(error);
-    }
+export class WebWalletError extends Error {
+  constructor ({
+    originalError,
+    customErrorMessage,
+    reportToSentry,
+    reportToUi
+  }) {
+    super(originalError.message);
+    this._originalError = originalError;
+    this._customErrorMessage = customErrorMessage;
+    this._reportToSentry = reportToSentry;
+    this._reportToUi = reportToUi;
   }
 
-  async sanitizeError (error) {
-    // user sign rejection from metamask
+  report (dispatchMethod) {
     if (
-      (error.message && error.message.toLowerCase().includes('user denied')) ||
-      (error.message && error.message.toLowerCase().includes('user rejected'))
+      noLogErrors.find(i => this._originalError.message && this._originalError.message.toLowerCase().includes(i)) ||
+      this._originalError.code === -32000
```

Comment:
i think we could define a meaningful variable to represent code like this (-32000)